modules/search/instant-search/lib/api.js

```diff
@@ -37,22 +39,80 @@ export function buildFilterAggregations( widgets = [] ) {
 	return aggregation;
 }
 
+const DATE_REGEX = /(\d{4})-(\d{2})-(\d{2})/;
+function generateDateRange( query, type ) {
+	// NOTE: This only supports a single date query at this time
+	const input = Array.isArray( query ) && query[ 0 ];
+
+	let year, month;
+	if ( type === 'year' ) {
+		[ , year, , ] = input.match( DATE_REGEX );
+	}
+
+	if ( type === 'month' ) {
+		[ , year, month ] = input.match( DATE_REGEX );
+	}
+
+	if ( month ) {
+		return { startDate: `${ year }-${ month }-01`, endDate: `${ year }-${ +month + 1 }-01` };
+	}
+	if ( year ) {
+		return { startDate: `${ year }-01-01`, endDate: `${ +year + 1 }-01-01` };
+	}
+	return { startDate: '', endDate: '' };
+}
+
 function buildFilterObject( filterQuery ) {
 	if ( ! filterQuery ) {
 		return {};
 	}
 
 	const filter = { bool: { must: [] } };
-	if ( Array.isArray( filterQuery.post_types ) && filterQuery.post_types.length > 0 ) {
+	if ( isLengthyArray( filterQuery.post_types ) ) {
 		filterQuery.post_types.forEach( postType => {
 			filter.bool.must.push( { term: { post_type: postType } } );
 		} );
 	}
-	if ( Array.isArray( filterQuery.post_tag ) && filterQuery.post_tag.length > 0 ) {
+	if ( isLengthyArray( filterQuery.post_tag ) ) {
 		filterQuery.post_tag.forEach( tag => {
 			filter.bool.must.push( { term: { 'tag.slug': tag } } );
 		} );
 	}
+	if ( isLengthyArray( filterQuery.month_post_date ) ) {
```

Comment:
The following section feels like it could be refactored to use less code by iterating through a list of the keys on filterQuery and then setting the appropriate field in the range filter. Unsure if that is worth it, but figured I'd mention it.

Reply:
I thought about doing so, but figured we could optimize this in the future if we found it necessary.